cilium-health/launch/endpoint.go

```diff
@@ -347,11 +347,11 @@ func LaunchAsEndpoint(baseCtx context.Context, owner regeneration.Owner, n *node
 
 	// Initialize the health client to talk to this instance. This is why
 	// the caller must limit usage of this package to a single goroutine.
-	client, err := probe.NewClient("http://" + net.JoinHostPort(healthIP.String(), fmt.Sprintf("%d", healthDefaults.HTTPPathPort)))
-	if err != nil {
+	client := &Client{host: "http://" + net.JoinHostPort(healthIP.String(), fmt.Sprintf("%d", healthDefaults.HTTPPathPort))}
+	if err = client.PingEndpoint(); err != nil {
```

Comment:
This initial ping is probably the cause for failing tests (preflight checks). I didn't see it locally when running the K8s suite. Note that NewClient only returned errors for incorrect URLs so it never checked the ability to establish a connection before.